src/esp/physics/CollisionGroupHelper.cpp

```diff
@@ -26,36 +26,45 @@ int CollisionGroupHelper::getMaskForGroup(CollisionGroup group) {
   switch (group) {
     case CollisionGroup::Default:
       // everything
-      return -1;
+      return -1 & ~int(CollisionGroup::Noncollidable);
 
     case CollisionGroup::Static:
-      // everything but kinematic
-      return int(-1) & ~int(CollisionGroup::Kinematic);
+      // everything but [kinematic, static]
+      return int(-1) & ~int(CollisionGroup::Kinematic) &
+             ~int(CollisionGroup::Static) & ~int(CollisionGroup::Noncollidable);
 
     case CollisionGroup::Kinematic:
-      // everything but static
-      return int(-1) & ~int(CollisionGroup::Static);
+      // everything but [kinematic, static]
+      return int(-1) & ~int(CollisionGroup::Kinematic) &
+             ~int(CollisionGroup::Static) & ~int(CollisionGroup::Noncollidable);
```

Comment:
Should kinematic objects not collide with other kinematic objects? Otherwise, why have both kinematic group and static group?

Reply:
kinematic objects shouldn't collide with other kinematic objects, because neither require a collision response.

Since they are conceptually different (kinematic objects move; static objects don't), I'm inclined to keep them as separate groups. I can imagine a day when we want selective collision, for example, a raycast that only hits static objects or only kinematic.

Reply:
I think being able to have collision location information is useful on demand regardless of the nature of the object, but i agree that it would be sub-optimal as default behavior for k-k since, as you said, no force response is appropriate.